src/pilot/assign_role.py

```diff
@@ -308,11 +308,13 @@ def define_controller_logical_disks(drac_client, raid_controller_name):
     raid_10_logical_disk = define_single_raid_10_logical_disk(
         drac_client, raid_controller_name)
 
-    if raid_10_logical_disk is None:
-        return None
-
     logical_disks = list()
-    logical_disks.append(raid_10_logical_disk)
+    if raid_10_logical_disk is None:
+        return logical_disks
```

Comment:
I'm pretty sure that on the line above, you need to:
return None
otherwise, assign_role won't exit with an error.  Can you make this change and test the error case to make sure this works correctly?  To test, you would need to hack define_single_raid_10_logical_disk() so that number_physical_disks=0 here:
https://github.com/dsp-jetpack/JetPack/blob/master/src/pilot/assign_role.py#L337
In this case, you should see a critical error logged here:
https://github.com/dsp-jetpack/JetPack/blob/master/src/pilot/assign_role.py#L373
and assign_role should exit with a 1 exit code here:
https://github.com/dsp-jetpack/JetPack/blob/master/src/pilot/assign_role.py#L1530